src/Bot.spec.ts

```diff
@@ -4,12 +4,20 @@ import { Bot } from './Bot'
 
 vi.mock('@/config')
 
+vi.mock('../../src/Bot.js', () => {
+  const Bot = vi.fn()
+  Bot.prototype.InitAndStart = vi.fn()
+
+  return { Bot }
+})
```

Comment:
the `Bot` class is a subject under test. please don’t mock it.

if you mock it, **we end up testing the mock** instead of testing the thing we want to test 😂. therefore, please be very mindful when deciding to use a mock

by the way i recommend reading this chapter from the book _Software Engineering at Google_: https://abseil.io/resources/swe-book/html/ch13.html

> Although mocking frameworks facilitate easier usage of test doubles, they come with some significant caveats given that their overuse will often make a codebase more difficult to maintain.